packages/imperative/src/imperative/__tests__/plugins/cmd/install/install.handler.unit.test.ts

```diff
@@ -12,6 +12,9 @@
 /* eslint-disable jest/expect-expect */
 import Mock = jest.Mock;
 
+let expectedVal: unknown;
+let returnedVal: unknown;
```

Comment:
Not sure we have many variables being defined prior to `import` statements.

I assume we do this because the `jest.doMock(...)` block needs them.

If that's the case (and this is just me being picky), could we move them both (variable definitions and `jest.doMock` block) below imports.

If I remember correctly, only `jest.mock` is treated specially and has to be called very early in the file. However, I believe we can call `jest.doMock` at any point.

This bit of [Jest doc](https://jestjs.io/docs/jest-object#jestdomockmodulename-factory-options) explains how using `babel-jest` hoists all `jest.mock` calls, and that we can use `jest.doMock` to avoid this behavior.

---

That being said, we we need to mock the `path` module before importing any libraries, then I'm fine to keep both (definitions and `doMock` block) where they are now.

Reply:
If definition of the two variables are moved after the imports, the test script fails with this error:
```
    ReferenceError: Cannot access 'expectedVal' before initialization
       > 31 |             if (path[0] == expectedVal) {
            |                            ^
```

If the related doMock is then also moved after the definition of the variables, then a test fails with the following error:
```
    Expected: "/some/test/directory/prod-plugins.json"
    Received: "C:\\ourstuff\\repos\\zowe-cli\\prod-plugins.json"

    Number of calls: 1

      172 |      */
      173 |     const wasReadFileSyncCallValid = (expectedFile: string) => {
    > 174 |         expect(mocks.readFileSync).toHaveBeenCalledWith(
          |                                    ^
      175 |             expectedFile
      176 |         );
      177 |     };

```

Instead of providing a test value, the script now seems to return a path to the root of the tester's source directory, which is obviously not what should happen.

Since the location of the two variables was part of a commit from @awharn, he may have some additional insight. Otherwise, I suggest that we leave the working test the way it is.

Reply:
I have no further insights, just experienced the same behaviors you found, @gejohnston. I wanted to fix the tests and get them back into a working order, and the simplest solution seemed to be moving the let statements to above the imports. I did not put much effort into the changes aside from making sure the tests passed, and the changes made sense. If we want to re-evaluate how the tests are being run and where we are defining variables, perhaps that would be good for another issue (when we are less time constrained). The main purpose of these changes were to just get things working ahead of V3 so we can properly test other changes that we plan on making later without worrying about these tests polluting the results. :)

Reply:
We can leave this test file (and possibly others like this) alone.
"_If it isn't broken, don't fix it!_" 😋 

I do appreciate you (@gejohnston) for spending some time to evaluate each and every single comment I made 😋